coordinator/internal/authority/authority_test.go

```diff
@@ -50,13 +53,52 @@ func TestSNPValidateOpts(t *testing.T) {
 
 // TODO(burgerdev): test ValidateCallback and GetCertBundle
 
+func TestBadStoreWatcherIsRestarted(t *testing.T) {
+	require := require.New(t)
+	fs := afero.NewBasePathFs(afero.NewOsFs(), t.TempDir())
+	store := &badStore{
+		Store: history.NewAferoStore(&afero.Afero{Fs: fs}),
+	}
+	ch := make(chan []byte)
+	store.ch.Store(&ch)
+	hist := history.NewWithStore(slog.Default(), store)
+	reg := prometheus.NewRegistry()
+	clock := testingclock.NewFakeClock(time.Now())
+	a := New(hist, reg, slog.Default(), clock)
+	t.Cleanup(a.Stop)
+
+	require.Eventually(func() bool { return store.n.Load() == 1 }, time.Second, 10*time.Millisecond, "Authority should call store.Watch after init")
```

Comment:
There's apparently a debate/issues with the async testify functions: https://github.com/stretchr/testify/pull/1657. Just as an fyi. Stumbled over this when searching for the implementation of https://github.com/stretchr/testify/issues/1087.

Reply:
Thanks, that's useful to keep in mind. For the usage at hand, I believe we are fine because the trigger for all of these issues seems to be a func argument that runs longer than the timeout, and with timeouts >10ms we should be safe when the func does only trivial local computation.